spack/tests/pass/pr-1105/example-7/output/entry.js

```diff
@@ -1,4 +1,16 @@
-const isMain = false;
-const modUrl = "$DIR/tests/pass/pr-1105/example-7/input/f.js";
+const isMain = {
+    url: "$DIR/tests/pass/pr-1105/example-7/input/f.js",
+    main: false
+}.main;
+const modUrl = {
+    url: "$DIR/tests/pass/pr-1105/example-7/input/f.js",
+    main: false
+}.url;
 console.log(isMain, modUrl);
-console.log(import.meta.main, "$DIR/tests/pass/pr-1105/example-7/input/entry.js");
+console.log(({
+    url: "$DIR/tests/pass/pr-1105/example-7/input/entry.js",
+    main: import.meta.main
+}).main, ({
+    url: "$DIR/tests/pass/pr-1105/example-7/input/entry.js",
+    main: import.meta.main
+}).url);
```

Comment:
WIP: I'd like this to emit like this:
```js
const import_meta_1 = {
    url: "$DIR/tests/pass/pr-1105/example-7/input/f.js",
    main: false
};
const isMain = import_meta_1.main;
const modUrl = import_meta_1.url;
console.log(isMain, modUrl);
const import_meta_2 = {
    url: "$DIR/tests/pass/pr-1105/example-7/input/entry.js",
    main: import.meta.main
};
console.log(import_meta_2.main, import_meta_2.url);
```
as I originally suggested in https://github.com/denoland/deno/issues/8088. I left a TODO in `merge.rs`. Since it's a bit more involved, I'd like to leave it for another PR.